pkg/kubelet/container/helpers.go

```diff
@@ -107,6 +107,23 @@ func HashContainer(container *v1.Container) uint64 {
 	return uint64(hash.Sum32())
 }
 
+// HashContainerImmutableFields returns the hash of the container with mutable fields zero'd out.
+func HashContainerImmutableFields(container *v1.Container) uint64 {
+	// InPlacePodVerticalScaling enables mutable Resources field.
+	// Changes to this field may not require container restart depending on policy.
+	// Compute hash over fields that are immutable with & without InPlacePodVerticalScaling
+	// NOTE: This is needed during alpha and beta so that containers using Resources but
+	//       not subject to In-place resize are not unexpectedly restarted when
+	//       InPlacePodVerticalScaling feature-gate is toggled.
+	//TODO(vinaykul): Remove this in GA+1 and make HashImmutableFields to become Hash.
+	hashImmutableFields := fnv.New32a()
+	containerCopy := container.DeepCopy()
+	containerCopy.Resources = v1.ResourceRequirements{}
```

Comment:
So there are 4 other fields that are mutable on the pod spec: [container image, initcontainer image, activeDeadlineSeconds and tolerations](https://kubernetes.io/docs/concepts/workloads/pods/#pod-update-and-replacement). We might want to pick a different name for this function to be clear about that, since it doesn't exclude all mutable fields.

Reply:
Good point. I'll just name it HashContainerWithoutResources and make this very explicit for this feature.

Reply:
fixed in latest commit

Reply:
Fixed a while ago. Resolving convo.